ui-v2/patches/ember-cli-code-coverage+1.0.0-beta.8.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/ember-cli-code-coverage/index.js b/node_modules/ember-cli-code-coverage/index.js
+index 6d84d18..81ef6fd 100644
+--- a/node_modules/ember-cli-code-coverage/index.js
++++ b/node_modules/ember-cli-code-coverage/index.js
+@@ -65,7 +65,7 @@ module.exports = {
+         )
+           .filter(Boolean)
+           .map(getPlugins)
+-          .forEach((plugins) => plugins.push([IstanbulPlugin, { exclude, include }]));
++          .forEach((plugins) => plugins.push([IstanbulPlugin, { include: ['consul-ui/utils/**/*','consul-ui/search/**/*'] }]));
```

Comment:
This appears to be configuration that could be passed in at the point of use rather than hardcode-patched like this...

Reply:
Hey @mikemorris 

Looks like you went down the same rabbit hole I did back when I did this in #7027!

> We also do a little hack to restrict the coverage report to only include what we want it to include as there doesn't seem to be a way to exclude files that have already been included by the addon.

Sorry! I'll add a bit more color here like you suggested so nobody goes down the same path again:

The [config file](https://github.com/kategengler/ember-cli-code-coverage#configuration) you link to above doesn't support the `include` option in its configuration (and therefore they don't document it at all). The option is supported/documented by npm package that actually does the work (https://github.com/istanbuljs/nyc#common-configuration-options) the ember addon just prevents you from using these options.

If you look at the javascript for the addon:

https://github.com/kategengler/ember-cli-code-coverage/blob/d6e2262923e7caeeb383fb6ccdf7b46e227f5715/index.js#L58-L69

the `includes` isn't pulled from the config file, only `excludes` is

https://github.com/kategengler/ember-cli-code-coverage/blob/d6e2262923e7caeeb383fb6ccdf7b46e227f5715/index.js#L221

So if you add the `includes` option from your PR:

https://github.com/hashicorp/consul/blob/9da4a1117d6f9a6cd72e8fcf9bda6951dde52ee9/ui-v2/config/coverage.js#L1-L3

It doesn't do anything.

The ember addon includes/hardcodes in all of the files that the author of the addon assumes you want, and that seems to mean that the `excludes` option has no effect on them (it won't exclude files you've already explicitly included).

The problem here is that we have different needs to the author of the addon and the addon removes the ability to configure istanbul how we'd like.

I vaguely remember trying an `.istanbul.yml` back when I did this, and I also could not get it to work which I think is due the ember addon hardcoding its `includes` on top of it. It seems to remove the ability to use an istanbul configuration file to do this also. Just to note here, there is a chance this may work but after trying for quite a while, I gave up figuring I was hitting the same problem as above and decided to move on. If you figure this out, then please lemme know!

As I'd tried absolutely everything else I could think of, I chose to patch as a last resort as I didn't wwant to get into the weeds of a PR to the addon.

At the time of the PR, I went with a straightforward make target (in the Consul UI I try to avoid writing scripts straight into the JSON in the package.json file as it just gets messy). I noticed that yarn had added the `patch` command and saw the `patch-package` the npm package and figured it would be clean and better to use this more common approach, hence this PR, which as we found out Friday, solves the linux/macOS `sed` differences in the makefile approach.

I think since going through this with you though I've realised that instead of trying to answer the question:

> How do I do this?

I've asked myself the question:

> Do I actually need to this?

And the answer to that second question is "no, I probably don't"

So if it sounds ok to you, I'm going to change up this PR to just remove the patching both using `patch-package` and the original make target, I don't mind the report listing everything, I can just pick out the things I'm interested in at the time.

Lemme know if that works and I'll go ahead and fix up this PR.

Reply:
After talking with @johncowen and poking at the upstream code a bit, apparently only `excludes` config is supported by the `ember-cli-code-coverage` and `lib-istanbul` libraries. Is the intent of this patch to actually change the measurements to prevent inflated metrics from full-stack tests (passing `--filter=Unit` in the `test:coverage` script is handling this already, measurements are identical without this patch) or only clean up the coverage report by ignoring paths considered to be irrelevant?

<img width="1425" alt="Screen Shot 2020-04-26 at 6 29 17 PM" src="https://user-images.githubusercontent.com/1149913/80321562-f96dac00-87eb-11ea-80bf-0da48a0876ec.png">
<img width="1422" alt="Screen Shot 2020-04-26 at 6 29 29 PM" src="https://user-images.githubusercontent.com/1149913/80321563-fa064280-87eb-11ea-9751-4f9be14a1294.png">

Reply:
Yeah exactamundo, its just so I can find what I'm interested in easier, I suppose its noise reduction.

I'll fix up the PR in a sec, or I might just close this and do another.